tests/setup-validators.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+set -eux
+# your baseledger binary name
+BIN=baseledgerd
+
+CHAIN_ID="baseledger"
+
+NODES=3
+
+ALLOCATION="10000000000stake,10000000000worktoken"
+
+# first we start a genesis.json with validator 1
+# validator 1 will also collect the gentx's once gnerated
+BASELEDGER_HOME="--home /validator"
+VALIDATOR_CONTAINER_BASE_NAME="baseledger-validator-container"
+STARTING_VALIDATOR_CONTAINER_NAME=$VALIDATOR_CONTAINER_BASE_NAME"1"
+docker exec $STARTING_VALIDATOR_CONTAINER_NAME  $BIN init $BASELEDGER_HOME validator --chain-id=$CHAIN_ID
+
+
+## Modify generated genesis.json to our liking by editing fields using jq
+## we could keep a hardcoded genesis file around but that would prevent us from
+## testing the generated one with the default values provided by the module.
+
+# add in denom metadata for both native tokens
+docker exec $STARTING_VALIDATOR_CONTAINER_NAME jq '.app_state.bank.denom_metadata += [{"name": "Work Token", "symbol": "WRK", "base": "worktoken", "description": "A test token for paying work", "denom_units": [{"denom": "worktoken", "exponent": 0}]},{"name": "Stake Token", "symbol": "STK", "base": "stake", "description": "A staking test token", "denom_units": [{"denom": "stake", "exponent": 0}]}]' /validator/config/genesis.json > /metadata-genesis.json
+docker cp /metadata-genesis.json $STARTING_VALIDATOR_CONTAINER_NAME:/metadata-genesis.json
+
+# a 60 second voting period to allow us to pass governance proposals in the tests
+docker exec $STARTING_VALIDATOR_CONTAINER_NAME jq '.app_state.gov.voting_params.voting_period = "60s"' /metadata-genesis.json > /edited-genesis.json
+docker cp /edited-genesis.json $STARTING_VALIDATOR_CONTAINER_NAME:/edited-genesis.json
+
+docker exec $STARTING_VALIDATOR_CONTAINER_NAME mv /edited-genesis.json /genesis.json
```

Comment:
does this mean we can edit genesis here for more params changes (check BAS-140, we can turn off inflation here i assume? not relevant for tests i guess)

Reply:
Yes